helix-view/src/editor.rs

```diff
@@ -242,6 +242,13 @@ pub struct LspConfig {
     pub auto_signature_help: bool,
     /// Display docs under signature help popup
     pub display_signature_help_docs: bool,
+    /// Time in milliseconds that the editor will wait for the lsp to shutdown before exiting
+    /// Set to 0 for instant. Defaults to 3000.
+    #[serde(
+        serialize_with = "serialize_duration_millis",
+        deserialize_with = "deserialize_duration_millis"
+    )]
+    pub shutdown_timeout: Duration,
```

Comment:
This should be a config option per-language-server (here: https://docs.helix-editor.com/languages.html#language-server-configuration) rather than editor-wide